### PR TITLE
Update wast to 6.0.0

### PR DESCRIPTION
**Pull request by abrown**
Due to API changes in wast, this removes all AssertReturn* directives and expands the functionality the single AssertReturn directive with the `nan:canonical` and `nan:arithmetic` patterns. See https://github.com/bytecodealliance/wat/pull/42 and https://github.com/WebAssembly/spec/pull/1104 for context.

### CODE REVIEW

**Comment by abrown** [comment]
Let's merge https://github.com/bytecodealliance/wasmtime/pull/798 first and I can rebase on that.

File: crates/wast/src/wast.rs [hunk 1]
```diff
@@ -542,15 +398,78 @@ fn is_arithmetic_f64_nan(bits: u64) -> bool {
...
+        V128Pattern::I8x16(_)
+        | V128Pattern::I16x8(_)
+        | V128Pattern::I32x4(_)
+        | V128Pattern::I64x2(_) => &actual.to_le_bytes() == to_le_bytes(expected).as_slice(),
+        V128Pattern::F32x4(expected_lanes) => {
+            let actual_lanes = vec![
+                extract_lane_as_u32(&actual, 0).expect("unable to extract lane"),
+                extract_lane_as_u32(&actual, 1).expect("unable to extract lane"),
+                extract_lane_as_u32(&actual, 2).expect("unable to extract lane"),
+                extract_lane_as_u32(&actual, 3).expect("unable to extract lane"),
```
Comment (abrown) [inline crates/wast/src/wast.rs]: @alexcrichton, we could get rid of the `expect`s here if we change `extract_lane_as_*` to not return a `Result` (why is that needed?).

File: crates/wast/src/wast.rs [hunk 2]
```diff
@@ -542,15 +398,78 @@ fn is_arithmetic_f64_nan(bits: u64) -> bool {
...
+        NanPattern::Value(expected_value) => actual == expected_value.bits,
+    }
+}
+
+fn to_le_bytes(v: &V128Pattern) -> Vec<u8> {
+    match v {
+        V128Pattern::I8x16(w) => w.iter().map(|&i| i as u8).collect(),
+        V128Pattern::I16x8(w) => w.iter().flat_map(|i| i.to_le_bytes().to_vec()).collect(),
+        V128Pattern::I32x4(w) => w.iter().flat_map(|i| i.to_le_bytes().to_vec()).collect(),
+        V128Pattern::I64x2(w) => w.iter().flat_map(|i| i.to_le_bytes().to_vec()).collect(),
```
Comment (abrown) [inline crates/wast/src/wast.rs]: I feel like there has to be some cool Rust way to just use arrays (not `Vec`s) here but I don't know it yet.

File: crates/wast/src/wast.rs [hunk 3]
```diff
@@ -542,15 +398,78 @@ fn is_arithmetic_f64_nan(bits: u64) -> bool {
...
+        _ => panic!("Unable to convert a float pattern to bytes (may contain NaN patterns)"),
```
Comment (abrown) [inline crates/wast/src/wast.rs]: Should this be a `bail!` with corresponding `Result`s in a few places?

File: crates/wast/src/wast.rs (hunk 1 above)
Comment (alexcrichton) [reply crates/wast/src/wast.rs]: Oh I think it's fine to remove the `Result`, I think that was an artifact from a previous time

File: crates/wast/src/wast.rs (hunk 2 above)
Comment (alexcrichton) [reply crates/wast/src/wast.rs]: Heh if such a cool way exists I am also not aware of it :)

File: crates/wast/src/wast.rs (hunk 3 above)
Comment (alexcrichton) [reply crates/wast/src/wast.rs]: Yeah where possible we want to use `Result` to improve the error messages coming out of failed or unsupported tests
